src/helics/common/addTargets.hpp

```diff
@@ -51,13 +58,32 @@ void addTargets(const Json::Value& section, std::string targetName, Callable cal
         } else {
             callback(targets.asString());
         }
+        found = true;
     }
     if (targetName.back() == 's') {
         targetName.pop_back();
         if (section.isMember(targetName)) {
             callback(section[targetName].asString());
+            found = true;
         }
     }
+    return found;
+}
+
+template<typename Block, typename Callable>
+void addTargetVariations(const Block& section,
+                         std::string name1,
+                         std::string name2,
+                         Callable callback)
```

Comment:
Should make sure the variants searched for are documented where they might affect a user, including dropping the "s" at the end of a target name. Using similar names could get confusing if a user notices that things still work even though one place has an extra "s" at the end of the name, and if someone else wrote the config file could raise questions on if the behavior was intended.

Reply:
the basic rule was that [targets] could be an array,  [target] had to be a singular element